jina/clients/base/http.py

```diff
@@ -212,3 +223,30 @@ def _result_handler(result):
                 if self.show_progress:
                     p_bar.update()
                 yield resp
+
+    async def _get_streaming_results(
+        self,
+        on: str,
+        inputs: 'Document',
+        parameters: Optional[Dict] = None,
+        return_type: Type[Document] = Document,
+        timeout: Optional[int] = None,
+        **kwargs,
+    ):
+        proto = 'https' if self.args.tls else 'http'
+        url = f'{proto}://{self.args.host}:{self.args.port}/{on.strip("/")}'
```

Comment:
this `on` is recomputed based on the `logic` used in `post` where it checks the OpenAPI.json to detect if the endpoint is there and otherwise it goes to default or something like that?

Reply:
it's actually inspired from this section: https://github.com/jina-ai/jina/blob/463eff659eeb5ac27be7fa1c569550ee546e3796/jina/clients/base/http.py#L148

Reply:
but what if the `on` is not a deployed endpoint? as it can happen for docarray v1

Reply:
Check the 3 if elses here:

```python
            if endpoint != '' and endpoint in self._endpoints:
                url = f'{proto}://{self.args.host}:{self.args.port}/{on.strip("/")}'
            elif has_default_endpoint:
                url = f'{proto}://{self.args.host}:{self.args.port}/default'
            else:
                url = f'{proto}://{self.args.host}:{self.args.port}/post'
```

Reply:
IMO, it's best to not mutate the endpoint that the user specifies. If it's not a deployed endpoint, it will fail.
I changed to the following logic: if the endpoint is in self._endpoints or there are no default endpoints, then use the endpoint as it is, otherwise it will use the default endpoint.
It does not make sense to use `/post` in this case